### PR TITLE
Remove PendingSignRequests queue from personalSign

### PR DESCRIPTION
**Pull request by sebastiandelgado**
Addresses #1027

This PR removes the PendingSignRequests queue from SignMessage.

It will be merged into feature/remove-transactions-queue-1027 along with PRs to:

- remove PendingSignRequests from Transactor
 - remove PendingSignRequests from PersonalRecover
 - remove PendingSignRequests

### CODE REVIEW

File: api/backend.go [hunk 1]
```diff
@@ -316,12 +328,16 @@ func (b *StatusBackend) registerHandlers() error {
 		return hash.Hex(), err
 	})
 
-	rpcClient.RegisterHandler(params.PersonalSignMethodName, b.personalAPI.Sign)
+	rpcClient.RegisterHandler(params.PersonalSignMethodName, unsupportedMethodHandler)
 	rpcClient.RegisterHandler(params.PersonalRecoverMethodName, b.personalAPI.Recover)
```
Comment (adambabik) [inline api/backend.go]: Do you plan to create another PR for this one?

File: api/backend.go [hunk 2]
```diff
@@ -236,6 +238,16 @@ func (b *StatusBackend) SendTransaction(ctx context.Context, args transactions.S
 	return b.transactor.SendTransaction(ctx, args)
 }
 
+// SignMessage checks the pwd vs the selected account and passes on the metadata
+// to personalAPI for message signature
+func (b *StatusBackend) SignMessage(rpcParams personal.Metadata) sign.Result {
```
Comment (adambabik) [inline api/backend.go]: I would keep `Password` as a separate argument. It's gonna be easier to keep backward compatibility and it will make it clear which part is our extension.

**Review by adambabik — APPROVED** [review]

File: api/backend.go (hunk 1 above)
Comment (sebastiandelgado) [reply api/backend.go]: No. Addressed it on this PR

File: api/backend.go (hunk 2 above)
Comment (sebastiandelgado) [reply api/backend.go]: Addressed

File: services/personal/api.go [hunk 3]
```diff
@@ -22,37 +23,44 @@ var (
...
+// UnmarshalSignRPCParams puts the RPC params for `personal_sign` or `web3.personal.sign`
+// into SignParams
+func UnmarshalSignRPCParams(rpcParamsJSON string) (SignParams, error) {
+	var params SignParams
+	err := json.Unmarshal([]byte(rpcParamsJSON), &params)
+	return params, err
+}
+
+// UnmarshalRecoverRPCParams
+func UnmarshalRecoverRPCParams(rpcParamsJSON string) (RecoverParams, error) {
```
Comment (sebastiandelgado) [inline services/personal/api.go:48]: Unsure about whether or not we want to unmarshal the rpcParams (which come into library.go as `rpcParams *C.char` and are converted into a string with `C.GoString`). The other option would be to pass them on as a `[]interface{}`, but this seems cleaner. 

@flexsurfer @adambabik thoughts on the current interface?

**Comment by adambabik** [comment]
After rebase with the current develop, the e2e tests should pass.

File: services/personal/api.go (hunk 3 above)
Comment (adambabik) [reply services/personal/api.go:48]: The current solution looks fine to me.

**Review by mandrigin — APPROVED** [review]
